### ISSUE TITLE
Example simulation to demonstrate water balance

### ISSUE THREAD

**Issue by hut104** [APSIMInitiative/ApsimX#8803]
### Describe the new feature

@sme016 has raised the following need.

Is there an example of a simulation in the GUI or that you could send me that has a complete annual summary of the components of a water balance? It would be useful to have this an Example in the GUI so that it can be accessed anytime for other purposes too. If you like, I can help you work up an example. I had version of this in the Pinus Example, but it is only for the top depth and it needs checking as a having all the correct fluxes, and others are not to know it is there for general use; it would be good to have a stand-alone example of doing a full water balance.

 

The reporting code I am using so currently is as follows:

Sum of ([MicroClimate].PrecipitationInterception) from [Clock].StartOfYear to [Clock].EndOfYear as AnnInterception

Sum of ([Weather].Rain) from [Clock].StartOfYear to [Clock].EndOfYear as AnnRain

Sum of ([Eucalyptus].FineRoot.WaterUptake) from [Clock].StartOfYear to [Clock].EndOfYear as AnnTEucalyptus

Sum of ([SoilWater].Evaporation) from [Clock].StartOfYear to [Clock].EndOfYear as AnnEvap

(AnnTEucalyptus + AnnEvap + AnnInterception) as AnnET

[Soil].SoilWater.Runoff from [Clock].StartOfYear to [Clock].EndOfYear as AnnRunoff

[Soil].SoilWater.Drainage from [Clock].StartOfYear to [Clock].EndOfYear as AnnDrainage

[Soil].SoilWater.Es from [Clock].StartOfYear to [Clock].EndOfYear as AnnSWEvap

(AnnRain-AnnRunoff-AnnInterception-AnnTEucalyptus-AnnSWEvap-AnnDrainage) as SoilWaterBalance

 

However,

For evaporation, I am not sure I should be using [SoilWater].Evaporation or [SoilWater].Es to tell me the flux upwards from the soil into the atmosphere, i.e. from the  soil surface or surface of ‘surface organic matter’?
In all years the SoilWaterBalance is positive, rather than my expectation of it balancing to zero over approximately a year. So I don’t think I have calculated the water balance completely correctly.
Ideally this code should include later inflow and outflow, but I think there shouldn’t be any in this case. It would be good to include that in the reporting, but again I am not completely sure about how to specify it.

**Comment by BrianCollinss** [comment]
At the risk of being 200% wrong: to have SoilWaterBalance close to 0 for any period of time, should there be a SoilWaterSotrage kind of thing to show how much water still exists in the profile?

`(AnnRain-AnnRunoff-AnnInterception-AnnTEucalyptus-AnnSWEvap-AnnDrainage-SoilWaterSotrage) as SoilWaterBalance
`

**Comment by sme016** [comment]
Yes, but actually its the change in SoilWaterStorage (positive or negative).
[SoilWater].Drainage needs checking too, as the values seem extraordinarily high compared to the rest of the fluxes.
Later inflow and outflow (lateral drainage) should also be included.

**Comment by sme016** [comment]
My comment above about high values of drainage can be ignored, as I had a water table in the simulation.

**Comment by sme016** [comment]
Here's an example of water balance reporting and graphing that could be tidied up and included as an example in the GUI.
[Sims.zip](https://github.com/APSIMInitiative/ApsimX/files/14941692/Sims.zip)

It has reporting for at day, month, year and simulation time-steps, and balance is close to zero over the long time periods.  

However, I  still don't know if I should be using didn't [SoilWater].Evaporation or [SoilWater].Es as the evaporation flux from the ground surface, and I didn't know how to generate and include lateral flow. So please include those.

Also, you could add an error check, as balance as a percentage of infiltration, which I think will be just a fraction of percent over the whole simulation of a few years.
